### ISSUE TITLE
Tags are randomly surrounded by quotes

### ISSUE THREAD

**Issue by mardy** [500px/api-documentation#250]
Hi!
  I've uploaded a photo with the following tags: uno, due, tre, quattro, cinque, sei. It succeeded, but the tags which have been set on the photo are:
`uno` `"due"` `tre` `"quattro"` `cinque` `"sei"`
It looks like every other one has extra quotes around it.

**Comment by mck182** [comment]
Hey Alberto,

can you send the full request (the body) you're sending?

Cheers,
Martin

**Comment by mardy** [comment]
Sure, here it is (tokens redacted):
```
auto_activate=true&category=0&description=&name=&privacy=0&tags=%22uno%22,%22due%22,%22tre%22,%22quattro%22,%22cinque%22,%22sei%22&oauth_signature_method=HMAC-SHA1&oauth_version=1.0&oauth_consumer_key=AAA&oauth_token=BBB&oauth_timestamp=1513629238&oauth_nonce=966420642&oauth_signature=CCC
```

**Comment by mardy** [comment]
Actually, the tags are not visible in the link I sent you, but in my "manage photos" page I can see them: and "due", "quattro" and "sei" have double quotes around them.

**Comment by mck182** [comment]
I see, so just to make sure - you're trying to have
tags with quotes around them but only every other
tag ends up with the quotes, correct?

**Comment by mardy** [comment]
Correct, except that I don't want to have any quotes around tags. :-)

**Comment by mck182** [comment]
So what if you just set them without quotes?

```
auto_activate=true&category=0&description=&name=&privacy=0&tags=uno,due,tre,quattro,cinque,sei&oauth_signature_method=...
```

That wouldn't work?

**Comment by mardy** [comment]
I need to try. But then, can I add tags containing a comma? Or are there some restrictions on what characters can be used in tags?

**Comment by mck182** [comment]
> can I add tags containing a comma?

Not really, no. But then tags with a comma don't
make that much sense, no? Do you have some
examples in mind?

**Comment by mardy** [comment]
Well, in some cases they [are used](https://www.flickr.com/photos/rbpdesigner/6645563201/in/photolist-pio3pE-bjRACC-aGHDzM-p8gxyj-pQ4cmh-ETEnjo-piSL2v-88wmJp-drLSzG-E2Stg8-9S62qW-pYjC37-pYG28T-KQ7mpA-pGynVn-9pud5g-KhaieK-pWfpNb-iLEyXk-DgJK2h-b8ffpT-am1uak-D6DpbT-e7tMSG-pWS4Ss-aTj3Lc-nSDc8X-pYma8b-bdMFQH-gcBFKF-pYcgQM-w4iKfw-UyzTcu-ed1W9K-gooTa5-dC4YPn-pFP8RP-dvNamk-pFWrC6-scc7PX-pYv5UJ-e6h9jJ-RCJg5u-QkVDgG-pFN8zt-pYp1Y2-pFYPUe-pYka48-SHHGqs-p68zti/), but I agree it's not a typical use-case. The important thing is to know which characters are allowed and which are not. :-)

**Comment by mck182** [comment]
If you really need to have comma in a tag, you
can send an array of tags, eg.

```
auto_activate=true&category=0&description=&name=&privacy=0&tags[]=uno&tags[]=due,tre&tags[]=quattro&tags[]=cinque,sei&oauth_signature_method=...
```

should result in tags `uno`, `due,tre`, `quattro`, `cinque,sei`

But then again, I'd argue that if you need to have
a comma in a tag, those should be two standalone
tags. Like you have `Roma, Italy`, it should really
be `Roma`, `Italy`.

**Comment by mardy** [comment]
OK, I've tried it and it seems to be working. I can even upload tags with quotes in them, which is good. However, I tried to upload a tag `that's life` and it failed. This is the body:
```
auto_activate=true&category=0&description=&name=&privacy=0&tags=that%27s life&oauth_signature_method=HMAC-SHA1&oauth_version=1.0&oauth_consumer_key=AAA&oauth_token=BBB&oauth_timestamp=1513711992&oauth_nonce=1662534350&oauth_signature=CCC
```
